### ISSUE TITLE
NavigationActions navigates incorrectly when TabNavigator is nested in DrawerNavigator

### ISSUE THREAD

**Issue by baldurpan** [react-navigation/react-navigation#3104]
RESUBMITTING THIS ISSUE (Edit: See #2043)
Added Expo snack here which clearly replicates the issue: https://snack.expo.io/r1It-LsWG

I believe I have found a bug when nesting TabNavigator in DrawerNavigator when using the same screen in multiple tabs.
In this example I'm using the screen "ProfileScreen" in both Tab1 and Tab2 and in Tab3 I have a button that should navigate to Tab2 -> ProfileScreen. This works perfectly if I only use TabNavigator but when It's nested in a DrawerNavigator, the navigator sends me to Tab1 -> ProfileScreen.

Here's my example App.js
```
import React from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { DrawerNavigator, TabNavigator, StackNavigator, NavigationActions } from 'react-navigation';

const ProfileScreen = () => {
  return (
    <View>
      <Text>This is profile</Text>
    </View>
  )
}

const Tab1Navigator = StackNavigator({
  Main: {
    screen: () => {
      return (
        <View>
          <Text>Main screen in tab 1</Text>
        </View>
      )
    },
  },
  Profile: {
    screen: ProfileScreen,
  },
});

const Tab2Navigator = StackNavigator({
  Main: {
    screen: () => {
      return (
        <View>
          <Text>Main screen in tab 2</Text>
        </View>
      )
    },
  },
  Profile: {
    screen: ProfileScreen,
  },
});

const Tab3Navigator = StackNavigator({
  Main: {
    screen: (props) => {
      return (
        <View>
          <Text>Main screen in tab 3</Text>

          <Button title="Take me to profle in tab2" onPress={() => {
            props.navigation.dispatch(NavigationActions.navigate({
              routeName: 'Tab2',
              params: {},
              action: NavigationActions.navigate({
                routeName: 'Profile',
              }),
            }));
          }}/>
        </View>
      )
    },
  },
});

const Tabs = TabNavigator({
  Tab1: {
    screen: Tab1Navigator,
  },
  Tab2: {
    screen: Tab2Navigator,
  },
  Tab3: {
    screen: Tab3Navigator,
  },
}, {
  lazy: true,
  tabBarPosition: 'bottom',
  tabBarOptions: {
    showIcon: false,
    activeTintColor: '#e91e63',
  },
});

const Drawer = DrawerNavigator({
  Tabs: {
    screen: Tabs,
  },
}, {
  contentComponent: () => <View />
});

export default class App extends React.Component {
  render() {
    return <Drawer />
  }
}
```

### Current Behavior
Button in Tab3 using NavigationActions sends me to Tab1 -> ProfileScreen

### Expected Behavior
Button in Tab3 using NavigationActions should be sending me to Tab2 -> ProfileScreen

| software         | version
| ---------------- | -------
| react-navigation |  1.0.0-beta.21
| react-native     |  0.50.0
| node             | 8.9.1
| npm or yarn      | yarn 1.3.2

**Comment by kelset** [comment]
Thanks for reporting, I have mostly one question atm: Do you need to have the screen called the same way in all 3 navigators? I mean, you could attach them to the same component but call it in different ways, in order to "avoid" having the navigation send you to "the first one it sees" with that name.

I also suggest that you take a look at this issue because the "root" is the same: https://github.com/react-community/react-navigation/issues/444

btw you wrote

> RESUBMITTING THIS ISSUE

Which was the first iteration of this?

**Comment by baldurpan** [comment]
> Which was the first iteration of this?

Oh right, sorry. The first one is at #2043 

> Do you need to have the screen called the same way in all 3 navigators? I mean, you could attach them to the same component but call it in different ways, in order to "avoid" having the navigation send you to "the first one it sees" with that name.

Well yes, of course I _could_ attach them with different names, but that's kinda just avoiding my/the issue. The given example here is an extremely simplified version of it really.

In my real project, I have dozens of screens under the "Profile" section and I declare all those routes in a single file. I then require that file inside 4 different tabs since they are possibly accessible from there (F.e. in a notifications tab and in a posts feed tab).

Using unique names for all those routes in all different tabs would feed like an unnecessary code duplication, or in my case, a code quadruplication. :)

**Comment by kelset** [comment]
Gotcha, but in that case, I really suggest you read the approach proposed here:  #444

Anyway I'll keep this open, but I'm not sure how it would be fixable.

**Comment by baldurpan** [comment]
Yes thanks, I checked #444 and it describes a similar scenario. However, I don't really have any issues navigating within each tab.

The issue here is really with the sub-action in NavigationActions. F.e. when a user presses a notification in my app, I want to send him to my tab2 (Notifications tab) and from there into the Profile screen. But he always gets sent to the Profile screen inside tab1 (Posts feed tab).

You see what I mean?

**Comment by baldurpan** [comment]
I'm not really sure how NavigationActions.navigate() works under the hood, but it would sound pretty straight forward to send the parent routeName to that sub action so it would search inside that route first. :)

**Comment by kelset** [comment]
> The issue here is really with the sub-action in NavigationActions.navigate(). F.e. when a user presses a notification in my app, I want to send him to my tab2 (Notifications tab) and from there into the Profile screen. But he always gets sent to the Profile screen inside tab1 (Posts feed tab).

yeah that's my guess too.

Btw not sure it's strictly related, but can you check if these PRs would help with fixing your issue?

https://github.com/react-community/react-navigation/pull/2544

https://github.com/react-community/react-navigation/pull/3094 (this prob is not really strictly related)

**Comment by brentvatne** [comment]
hi there!

this is resolved now -- i updated to 1.3.1 and it works as expected: https://snack.expo.io/H1W1_rDuz

you'll just want to be careful that if you navigate to the profile, then go back to tab3 and navigate to profile again, it'll push a second profile screen. to handle this, use a key:

```js
action: NavigationActions.navigate({
   routeName: 'Profile',
   key: 'tab2-profile' // or some thing like the id
}),
```
